### PR TITLE
install_subdir: Also copy dangling symlinks

### PR DESCRIPTION
**Pull request by nirbheek**
Fixes https://github.com/mesonbuild/meson/issues/3914

### CODE REVIEW

**Comment by jpakkane** [comment]
Test failures seem relevant.

**Comment by nirbheek** [comment]
Hmm, I wonder how to make zipapp exclude some files. The way it is being built right now means that several MB of tests are included which bloats the size. This is also what is causing the test failure on Linux and macOS.

**Comment by nirbheek** [comment]
(closed it by accident, misclick)

I just realized that the test passes on VS because we don't actually run `meson install` there. Will fix that too.

**Comment by nirbheek** [comment]
> I just realized that the test passes on VS because we don't actually run meson install there. Will fix that too.

I opened https://github.com/mesonbuild/meson/issues/4028 for this, so that this PR can go into the stable release.

File: run_unittests.py [hunk 1]
```diff
@@ -3763,6 +3763,30 @@ def test_usage_external_library(self):
...
+        # the creation of zipapps. Create it dynamically and run the test by
+        # hand.
+        os.symlink('../../nonexistent.txt', 'test.txt')
+        try:
+            self.init(testdir)
+            self.build()
+            self.install()
+            link = os.path.join(self.installdir, 'usr', 'share', 'sub1', 'test.txt')
+            self.assertTrue(os.path.islink(link), msg=link)
+            self.assertFalse(os.path.isfile(link), msg=link)
```
Comment (jpakkane) [inline run_unittests.py]: Maybe this should do a `readlink` and verify that it points to the correct place?

Comment (nirbheek) [reply run_unittests.py]: Done.

**Comment by jpakkane** [comment]
Ok to merge once tests pass.

**Pull request merged**